Sources/SwiftSyntax/Documentation.docc/Contributing/ChangingSwiftSyntax.md

```diff
@@ -12,31 +12,27 @@ structures in sync, this project generates code using the Swift package
 ## Regenerating Files
 
 To re-generate the files after changing `CodeGeneration` run the `generate-swiftsyntax` 
-target of `CodeGeneration` and pass `path/to/swift-syntax/Sources` as the argument.
+target of `CodeGeneration`.
 
 On the command line, this would be
 ```bash
-swift run --package-path CodeGeneration generate-swiftsyntax Sources
+swift run --package-path CodeGeneration generate-swiftsyntax
 ```
 
-Or if you open the `CodeGeneration` package in Xcode, you can add the 
-`generate-swift syntax ../Sources` arguments using Product -> Scheme -> Edit Scheme…
-
 ## Adding and Removing Syntax Nodes
 
 The files containing the definition of all of the syntax nodes are available
-under the [SyntaxSupport][SyntaxSupport] directory. These files
+under the [SyntaxSupport](SyntaxSupport) directory. These files
 are roughly divided according to broad syntactic categories in the Swift
 programming language. That is, the syntax nodes for classes, structs, and actors
-are defined in `DeclNodes.py`, while the syntax nodes for string literals, 
-arrays, and tuples is defined in `ExprNodes.py`.
+are defined in `DeclNodes.swift`, while the syntax nodes for string literals, 
+arrays, and tuples is defined in `ExprNodes.swift`.
 
 To add a node to these files, it can be helpful to copy an existing node and 
 alter its definition to suit the needs of the new syntax being defined. A syntax
-node consists of a name, a kind, and a list of child nodes. The name of a node
-not only identifies it uniquely to Swift Syntax, it also provides a handle that
-other nodes can use to refer to it. The kind of a syntax node defines the
-class of syntax the node belongs to. All nodes are at least of the `Syntax`
+node consists of a kind, which also defines the node’s name, a base kind, and a list of 
```

Comment:
```suggestion
node consists of at least a kind (which also defines the node’s name), a base kind, and a list of child nodes
```

Is `nameForDiagnostics` required? If so maybe add that to the list as well. And do you think it would be useful to add a trailing comment to each below, or the documentation in `Node` is enough?

Reply:
`nameForDiagnostics` is not required. If `nil` it will use the `nameForDiagnostics` of the parent node.

I think the documentation of `Node` is enough. If we duplicate it and change it in the future, we will just forget to update this document again.